frontend/src/api/actions/init.js

```diff
@@ -0,0 +1,150 @@
+
+        if (newMockAPIValue === true) {
+            mockAPI.replaceGlobalFetch();
+        } else {
+            mockAPI.restoreGlobalFetch();
+        }
+    }
+    /* eslint-enable */
+};
+
+const updateGlobals = (getState, newMockAPIValue, newActiveSession) => {
+    const state = getState();
+    let globals = globalsSelector(state);
+
+    if (newMockAPIValue != null) {
+        globals = {
+            ...globals,
+            mockAPI: newMockAPIValue
+        };
+    }
+
+    if (newActiveSession) {
+        globals = {
+            ...globals,
+            activeSession: newActiveSession.id
+        };
+    }
+
+    return setGlobals(globals);
+};
+
+const generateListOfActions = (getState, stage, options) => {
+    // TODO: enable activeUserRole when setActvieUserRole mockAPI/API is implemented
+    const {
+        activeSession = null,
+        // activeUserRole = null,
+        mockAPI = null
+    } = options;
+
+    const onInitialization = [
+        {
+            stage: "toggleMockAPI",
+            function: toggleMockApi,
+            variables: [getState, mockAPI]
+        },
+        {
+            stage: "setActiveUser",
+            action: fetchActiveUser,
+            variables: []
+        },
+        // {
+        //     stage: "setActiveUserRole",
+        //     action: setActiveUserRole,
+        //     variables: [getState, activeUserRole]
+        // },
+        {
+            stage: "fetchSessions",
+            action: fetchSessions,
+            variables: []
+        }
+    ];
+
+    // Only send modify active session when
+    // activeSession is passed in
+    if (activeSession) {
+        onInitialization.push({
+            stage: "setActiveSession",
+            action: setActiveSession,
+            variables: [activeSession]
+        });
+    }
+
+    onInitialization.push({
+        stage: "updateActiveSessionGlobal",
+        action: updateGlobals,
+        variables: [getState, mockAPI, activeSession]
+    });
+
+    while (onInitialization.length > 0) {
+        if (onInitialization[0].stage === stage) {
+            break;
+        }
+
+        onInitialization.shift();
+    }
+    return onInitialization;
+};
+
+const fetchAdditionalData = [
+    fetchApplicants,
+    fetchApplications,
+    fetchAssignments,
+    fetchContractTemplates,
+    fetchInstructors,
+    fetchPositions
+];
+
+export function initFromStage(options, stage = "toggleMockAPI") {
+    return async (dispatch, getState) => {
+        const onInitialization = generateListOfActions(
+            getState,
+            stage,
+            options
+        );
+
+        const promises = [];
+        for (const step of onInitialization) {
+            if (step.action) {
+                promises.push(dispatch(step.action(...step.variables)));
+            } else {
+                step.function(...step.variables);
+            }
+        }
+
+        for (const fetchAction of fetchAdditionalData) {
```

Comment:
You're aiming at something too abstract. Make a big, long, ugly function that literally dispatches the actions in order.